internal/mode/static/telemetry/collector.go

```diff
@@ -98,6 +151,7 @@ func (c DataCollectorImpl) Collect(ctx context.Context) (Data, error) {
 	if err != nil {
 		return Data{}, fmt.Errorf("failed to collect NGF replica count: %w", err)
 	}
+	deploymentFlagOptions := collectDeploymentFlagOptions(c.cfg.Flags)
```

Comment:
Since the flags are static - they don't change during the runtime --  so we only need to collect them once.

It's better to parse them in the main package and then propagate parsed data in the Config.

This way we will localize dependency on pflag in the main package.

Reply:
Just double checking, if a user updates the NGF configuration such that it affects any of the flags, does NGF restart and thus rerun `createStaticModeCommand()` -- allowing for us to get the flag changes? 

> It's better to parse them in the main package and then propagate parsed data in the Config.

I agree with you that since the flags are static, the processing and logic doesn't need to be run each time the Collector collects, thanks for catching that. However, the idea of processing the flags in main seems a bit off to me and it feels like the functionality shouldn't be that high up. 

How does the idea of passing the flagset to the DataCollectorImpl, but doing the processing of parsing the flags in the constructor `NewDataCollectorImpl` so that it only runs once (when the DataCollector is created, in startmanager), but the logic is closer to the collector? 

Otherwise, how does passing the flagset through to the manager and doing the parsing logic in `StartManager`?

Reply:
> Just double checking, if a user updates the NGF configuration such that it affects any of the flags, does NGF restart and thus rerun createStaticModeCommand() -- allowing for us to get the flag changes?

yep

>  However, the idea of processing the flags in main seems a bit off to me and it feels like the functionality shouldn't be that high up.

it can be done in a package there if that is concern.
but, let's think about this way. what if we need to replace the flags library. if you pass the flags all the way down, you will need to change more parts of the code. However, if you parse the flags in main, and propagate the data, the data will not need to be changed.

similarly, we don't pass the flags to Manager to get configuration options. We parse the flags and put them into Config structs.

Reply:
okay that makes more sense, I'll do that!